examples/asr/speech_classification/frame_vad_infer.py

```diff
@@ -0,0 +1,266 @@
+        smoothing_pred_dir = generate_overlap_vad_seq(
+            frame_pred_dir=pred_dir,
+            smoothing_method=cfg.vad.parameters.smoothing,
+            overlap=cfg.vad.parameters.overlap,
+            window_length_in_sec=cfg.vad.parameters.window_length_in_sec,
+            shift_length_in_sec=cfg.vad.parameters.shift_length_in_sec,
+            num_workers=cfg.num_workers,
+            out_dir=cfg.smoothing_out_dir,
+        )
+        logging.info(
+            f"Finish generating predictions with overlapping input segments with smoothing_method={cfg.vad.parameters.smoothing} and overlap={cfg.vad.parameters.overlap}"
+        )
+        pred_dir = smoothing_pred_dir
+
+    # postprocessing and generate speech segments
+    logging.info("Converting frame level prediction to RTTM files.")
+    rttm_out_dir = generate_vad_segment_table(
+        vad_pred_dir=pred_dir,
+        postprocessing_params=cfg.vad.parameters.postprocessing,
+        frame_length_in_sec=frame_length_in_sec,
+        num_workers=cfg.num_workers,
+        use_rttm=cfg.vad.use_rttm,
+        out_dir=cfg.rttm_out_dir,
+    )
+    logging.info(
+        f"Finish generating speech semgents table with postprocessing_params: {cfg.vad.parameters.postprocessing}"
+    )
+
+    logging.info("Writing VAD output to manifest")
+    key_pred_rttm_map = {}
+    manifest_new = []
+    for entry in manifest_orig:
+        key = Path(entry['audio_filepath']).stem
+        entry['rttm_filepath'] = Path(os.path.join(rttm_out_dir, key + ".rttm")).absolute().as_posix()
+        if not Path(entry['rttm_filepath']).is_file():
+            logging.warning(f"Not able to find {entry['rttm_filepath']} for {entry['audio_filepath']}")
+            entry['rttm_filepath'] = ""
+        manifest_new.append(entry)
+        key_pred_rttm_map[key] = entry['rttm_filepath']
+
+    if not cfg.out_manifest_filepath:
+        out_manifest_filepath = "manifest_vad_output.json"
+    else:
+        out_manifest_filepath = cfg.out_manifest_filepath
+    write_manifest(out_manifest_filepath, manifest_new)
+    logging.info(f"Finished writing VAD output to manifest: {out_manifest_filepath}")
+
+    if cfg.get("evaluate", False):
+        auroc, report = eval_detection_error(
+            pred_dir=pred_dir,
+            key_labels_map=key_labels_map,
+            key_rttm_map=key_rttm_map,
+            key_pred_rttm_map=key_pred_rttm_map,
+            frame_length_in_sec=frame_length_in_sec,
+        )
+        DetER = report.iloc[[-1]][('detection error rate', '%')].item()
+        FA = report.iloc[[-1]][('false alarm', '%')].item()
+        MISS = report.iloc[[-1]][('miss', '%')].item()
+        logging.info(f"AUROC: {auroc:.4f}")
+        logging.info(f"DetER={DetER:0.4f}, False Alarm={FA:0.4f}, Miss={MISS:0.4f}")
+        logging.info(f"with params: {cfg.vad.parameters.postprocessing}")
+    logging.info("Done!")
+
+
+def load_input_manifest(cfg):
```

Comment:
and function like this could be moving to vad_utils

Reply:
done